pkg/document/json/rich_text.go

```diff
@@ -77,10 +78,15 @@ func (t *RichTextValue) AnnotatedString() string {
 
 // Split splits this value by the given offset.
 func (t *RichTextValue) Split(offset int) RGATreeSplitValue {
-	value := t.value
-	r := []rune(value)
-	t.value = string(r[0:offset])
-	return NewRichTextValue(t.attrs.DeepCopy(), string(r[offset:]))
+	grapheme := uniseg.NewGraphemes(t.value)
+
+	var split []string
+	for idx := 0; grapheme.Next(); idx++ {
```

Comment:
```suggestion
	for grapheme.Next() {
```
I think it doesn't need "i" here...! 🤔